### PR TITLE
fix fix_add fresh recall issue

### PR DESCRIPTION
**Pull request by MaggieQi**

### CODE REVIEW

File: AnnService/inc/Core/Common/WorkSpace.h [hunk 1]
```diff
@@ -135,54 +135,71 @@ namespace SPTAG
...
         public:
             DistPriorityQueue() {}
 
             void Resize(int size_) {
-                m_data.reset(new float[size_ + 1]);
-                for (int i = 0; i <= size_; i++) m_data[i] = MaxDist;
                 m_size = size_;
+                m_data.reset(new float[size_ + 1]);
+                
+                m_data[1] = MaxDist;
```
Comment (SkyInTheSea) [inline AnnService/inc/Core/Common/WorkSpace.h:150]: what's the usage of m_data{0], why don't you use m_data[0] to store MaxDist?

Comment (MaggieQi) [reply AnnService/inc/Core/Common/WorkSpace.h:150]: to make the tree traverse faster, the root index is better to be 1.

Comment (SkyInTheSea) [reply AnnService/inc/Core/Common/WorkSpace.h:150]: then m_data[0] is unused?

**Review by SkyInTheSea — DISMISSED** [review]

File: AnnService/inc/Core/Common/WorkSpace.h (hunk 1 above)
Comment (MaggieQi) [reply AnnService/inc/Core/Common/WorkSpace.h:150]: yes

Comment (SkyInTheSea) [reply AnnService/inc/Core/Common/WorkSpace.h:150]: could you consider to use clean m_data to reuse original array and avoid extra heap allocation?

File: AnnService/inc/Core/Common/WorkSpace.h [hunk 2]
```diff
@@ -135,54 +135,71 @@ namespace SPTAG
...
-            std::unique_ptr<float[]> m_data;
             int m_size;
+            std::unique_ptr<float[]> m_data;
+            int m_length;
             int m_count;
+            
         public:
             DistPriorityQueue() {}
 
             void Resize(int size_) {
```
Comment (SkyInTheSea) [inline AnnService/inc/Core/Common/WorkSpace.h:146]: may add some defense code when size_ <=0,
may consider optimization when size_ <= m_size.

File: AnnService/inc/Core/Common/WorkSpace.h [hunk 3]
```diff
@@ -135,54 +135,71 @@ namespace SPTAG
...
-                m_data.reset(new float[size_ + 1]);
-                for (int i = 0; i <= size_; i++) m_data[i] = MaxDist;
                 m_size = size_;
+                m_data.reset(new float[size_ + 1]);
+                
+                m_data[1] = MaxDist;
+                m_length = 1;
                 m_count = size_;
             }
             void clear(int count_) {
```
Comment (SkyInTheSea) [inline AnnService/inc/Core/Common/WorkSpace.h:154]: what's the difference of clear() and resize()

**Review by SkyInTheSea — APPROVED** [review]

File: AnnService/inc/Core/Common/WorkSpace.h (hunk 2 above)
Comment (MaggieQi) [reply AnnService/inc/Core/Common/WorkSpace.h:146]: resize is only used in initialization. we can add size_ > 0 check in the next submit.

File: AnnService/inc/Core/Common/WorkSpace.h (hunk 3 above)
Comment (MaggieQi) [reply AnnService/inc/Core/Common/WorkSpace.h:154]: clear() is used to avoid reallocate when count_ <= m_size.

**Pull request merged**